src/main/java/seedu/jarvis/model/ModelManager.java

```diff
@@ -312,6 +324,7 @@ public boolean hasTodo(Todo todo) {
     @Override
     public void addTodo(Todo todo) {
         addressBook.addTodo(todo);
+        updateTasksSummaryDetail();
```

Comment:
you can probably updateTasksSummaryDetail() at higher level so that better abstraction is followed, no need to write so many of the same functions that may appear confusing.